staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -797,13 +823,18 @@ func (c *Cacher) dispatchEvents() {
 	defer bookmarkTimer.Stop()
 
 	lastProcessedResourceVersion := uint64(0)
+	metricLabelValue := c.objectType.String()
 	for {
 		select {
 		case event, ok := <-c.incoming:
 			if !ok {
 				return
 			}
+			now := c.clock.Now()
+			cacherEventsCounter.WithLabelValues(metricLabelValue).Inc()
 			c.dispatchEvent(&event)
+			cacherEventsSentLatency.WithLabelValues(metricLabelValue).Observe(float64(c.clock.Since(now)/time.Microsecond) / 1000.0)
```

Comment:
What is the performance impact of adding that on critical path in heavily loaded large cluster?
I'm slighly feared about that.